portal-api/src/v1/controllers/eligibilityCriteria.controller.js

```diff
@@ -69,6 +69,13 @@ exports.update = async (req, res) => {
 
 exports.delete = async (req, res) => {
   const collection = await db.getCollection('eligibilityCriteria');
-  const status = await collection.deleteOne({ version: Number(req.params.version) });
-  res.status(200).send(status);
+  const { version } = req.params;
+  const versionNumber = Number(version);
+
+  if (typeof versionNumber === 'number') {
+    const status = await collection.deleteOne({ version: { $eq: versionNumber } });
+    res.status(200).send(status);
+  }
+
+  res.status(400).send('Invalid eligibility criteria version number');
```

Comment:
I think the `if (typeof versionNumber === 'number')` check will always be `true` here, because `Number(...)` always returns a number (if `version` wasn't a valid number then `Number(version)` will be `NaN` and `typeof NaN` is `number`)

Also, do we need a `return` after line 77?

Reply:
Have refactored.